numpy/core/src/common/array_assign.h

```diff
@@ -1,6 +1,8 @@
 #ifndef _NPY_PRIVATE__ARRAY_ASSIGN_H_
 #define _NPY_PRIVATE__ARRAY_ASSIGN_H_
 
+#include "numpy/arrayobject.h"
+
```

Comment:
Why the extra include?

Reply:
> Why the extra include?

It was to make the include order safe. However, it turns out that `.clang-format` provides a mechanism for automatically grouping and ordering  includes,  so I am going to formalize our include ordering.

Reply:
As far as I can tell this include is not moved, it is new to the file.

Reply:
> this include is not moved

True, it happened to sort after the needed `numpy/*` includes, so didn't cause a problem, but that was an accident. I'm going the put the `numpy/*` includes into their own group so they always come first and then we won't need to worry.